addons/mail/static/src/composer/composer.xml

```diff
@@ -90,6 +90,7 @@
                     imagesHeight="50"/>
             </div>
         </div>
+        <EmojiPicker t-if="env.isSmall" onSelect="(str) => this.addEmoji(str)" close="() => this.state.autofocus++" restorePositionOnSelect="false" storeScroll="useEmojiPickerStoreScroll()" className="state.showEmojiPicker ? 'o-show' : 'o-hide'"/>
```

Comment:
We should always mount the `EmojiPicker` and have it handle its visibility + internal state `storeScroll`.
By doing so, you won't have to define the `useEmojiPickerStoreScroll`, and you won't have to rely on `d-none` in the concern of Composer (this is an implementation detail of the `EmojiPicker`, and actually you could do this without `d-none`)

Reply:
Yes, but the problem is when we are using popover for the `EmojiPicker`, the component is always created and destroyed. This is the behavior of the popover hook. If we always have a popover but replace everything related to `onClose` with `onClick` and make the popover hidden when `onClick`, it works, but not the idea of how a component should be using a popover hook. that's why we need to store the position outside of the component.

Reply:
I really don't like how some code that is internal of emoji picker because bits outside of internal code of emoji picker.

Sure, we tricked by putting state in the component hook rather than `EmojiPicker`, and this helps having persistence of state when emoji picker is closed. That's the nice thing of component hook: it defines a feature rather than a component, so the feature can work without coupling it to sub-child lifecycle.

Component hooks have their limitations, so I get why you use component... But I think components that are pure showing of content like how `EmojiPicker` currently is, this prevents encapsulate all the logic of the feature, including when the content is not shown.

I really think this case should also work with `useEmojiPicker()`. I get it's not conventional to do this, but it's one of the 1st component where we want 2 different presentation in desktop/mobile, and I really dislike this code duplication + totally different approaches when it's just a change of container + leaking of internal features of EmojiPicker + added complexity of API of `EmojiPicker`.

The increase of complexity should stay inside the EmojiPicker, not outside.
Using the EmojiPicker should stays very straightforward to use.